forecasttools/data.py

```diff
@@ -0,0 +1,478 @@
+    if not os.path.isdir(save_directory):
+        raise NotADirectoryError(
+            f"The path {save_directory} is not a directory."
+        )
+    # check if the save (output) file already exists
+    file_save_path = os.path.join(save_directory, file_save_name)
+    if os.path.exists(file_save_path):
+        raise FileExistsError(f"The file {file_save_path} already exists.")
+    # create dataset(s) from NHSN data
+    else:
+        # check that a data file exists
+        if not os.path.exists(nhsn_dataset_path):
+            raise FileNotFoundError(
+                f"The file {nhsn_dataset_path} does not exist."
+            )
+        else:
+            # check that the loaded CSV has the needed columns
+            df_cols = pl.scan_csv(nhsn_dataset_path).columns
+            required_cols = [
+                "state",
+                "date",
+                "previous_day_admission_adult_covid_confirmed",
+                "previous_day_admission_influenza_confirmed",
+            ]
+            if len(set(required_cols).intersection(set(df_cols))) != len(
+                required_cols
+            ):
+                raise ValueError(
+                    f"NHSN dataset missing required columns: {set(required_cols) - set(required_cols).intersection(set(df_cols))}"
+                )
+            # fully load and save NHSN dataframe
+            df = pl.read_csv(nhsn_dataset_path)
+            # change date formatting to ISO8601
+            df = df.with_columns(df["date"].str.replace_all("/", "-"))
+            # pathogen specific df saving
+            if dataset == "COVID":
+                df_covid = (
+                    df.select(
+                        [
+                            "state",
+                            "date",
+                            "previous_day_admission_adult_covid_confirmed",
+                        ]
+                    )
+                    .rename(
+                        {
+                            "previous_day_admission_adult_covid_confirmed": "hosp"
+                        }
+                    )
+                    .sort(["state", "date"])
+                )
+                df_covid.write_csv(file_save_path)
+            if dataset == "flu":
+                df_flu = (
+                    df.select(
+                        [
+                            "state",
+                            "date",
+                            "previous_day_admission_influenza_confirmed",
+                        ]
+                    )
+                    .rename(
+                        {"previous_day_admission_influenza_confirmed": "hosp"}
+                    )
+                    .sort(["state", "date"])
+                )
+                df_flu.write_csv(file_save_path)
+            print(
+                f"The file {file_save_name} has been created at {save_directory}."
+            )
+
+
+def make_nhsn_fitted_forecast_idata(
```

Comment:
separate into different functions:
1. Create the idata
2. Plot the idata
3. save the idata

Reply:
This makes some sense. The reason this code is in one function is because my expectation was that it would seldomly be ran by users. Counterpoint: the utility of the code is not uniform across function, so I agree that having three separate functions would be good.

Reply:
The function `make_nhsn_fitted_forecast_idata` has been removed for the time being. I will document the `idata` creation process and possibly re-introduce the models in a future vignette. Should this occur, I will split the function `make_nhsn_fitted_forecast_idata` into several functions, as you have requested. There will be plotting and saving utilities in a future PR.